### PR TITLE
bpo-47007: [doc] `str` special method lookup

### PR DESCRIPTION
**Pull request by itsvs**
Adds a note to the `str()` docs pointing out that `object.__str__()` follows special method lookup.

https://bugs.python.org/issue47007

https://bugs.python.org/issue47007

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@itsvs

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by JelleZijlstra** [comment]
This seems excessively long to me. I'd prefer just saying `type(object).__str__(object)`, as the bug suggested.

**Comment by itsvs** [comment]
sure, changed to just `type(object).__str__(object)` (do these docs prefer the 80-character cutoff? i didn't reformat)

File: Doc/library/stdtypes.rst [hunk 1]
```diff
@@ -1517,7 +1517,7 @@ multiple fragments.
    depends on whether *encoding* or *errors* is given, as follows.
 
    If neither *encoding* nor *errors* is given, ``str(object)`` returns
-   :meth:`object.__str__() <object.__str__>`, which is the "informal" or nicely
+   :meth:`type(object).__str__(object) <object.__str__>`, which is the "informal" or nicely
```
Comment (JelleZijlstra) [inline Doc/library/stdtypes.rst]:
```suggestion
   :meth:`type(object).__str__(object) <object.__str__>`,
   which is the "informal" or nicely
```
Leave the line within the length limits

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**